src/textual/_node_list.py

```diff
@@ -19,15 +19,13 @@ class NodeList:
     """
 
     def __init__(self) -> None:
-        self._node_refs: list[ref[DOMNode]] = []
-        self.__nodes: list[DOMNode] | None = []
+        self._nodes: list[DOMNode] = []
```

Comment:
This class has changed in purpose. It was a container that didn't hold on to references. Now it is a simpler type of container.

It could perhaps be a list, and may become one eventually. But it may also present a interface to the developer that prevents them from breaking things.